docs/spec/staking/spec-technical.md

```diff
@@ -154,76 +169,82 @@ Description:
+    StartSlashRatio  rational.Rat  // candidate slash ratio 
 }
 ``` 
-In the unbonding queue - the fraction of all historical slashings on
-that validator are recorded (`StartSlashRatio`). When this queue reaches maturity
-if that total slashing applied is greater on the validator then the
-difference (amount that should have been slashed from the first validator) is
-assigned to the amount being paid out.  
+
+TODO: Explain what is StartSlashRation.
 
 ### QueueElemReDelegate
 
+QueueElemReDelegate structure is used in the re-delegation queue. 
+
 ``` golang
 type QueueElemReDelegate struct {
 	QueueElem
-	Payout       Address  // account to pay out to
+	Payout       Address       // account to pay out to
     Shares       rational.Rat  // amount of shares which are unbonding
     NewCandidate crypto.PubKey // validator to bond to after unbond
 }
 ```
+Q: Why we need Payout address in the re-delegation element?
 
 ### Transaction Overview
 
 Available Transactions: 
  - TxDeclareCandidacy
- - TxEditCandidacy
- - TxLivelinessCheck
- - TxProveLive 
+ - TxEditCandidacy 
  - TxDelegate
  - TxUnbond 
  - TxRedelegate
+ - TxLivelinessCheck
+  - TxProveLive
 
 ## Transaction processing
 
 In this section we describe the processing of the transactions and the corresponding updates to the global state.
-For the following text we will use gs to refer to the GlobalState data structure, candidateMap is a reference to the 
-map[PubKey]Candidate, delegatorBonds is a reference to map[[]byte]DelegatorBond, unbondDelegationQueue is a 
-reference to the queue[QueueElemUnbondDelegation] and redelegationQueue is the reference for the 
-queue[QueueElemReDelegate]. We use tx to denote reference to a transaction that is being processed.      
+In the following text we will use `gs` to refer to the `GlobalState` data structure, 
+`unbondDelegationQueue` is a reference to the `queue[QueueElemUnbondDelegation]`, `reDelegationQueue` is the reference for the 
+`queue[QueueElemReDelegate]`. We use `tx` to denote a reference to a transaction that is being processed, and `sender` to
+denote the address of the sender of the transaction. We use function `loadCandidate(store, PubKey)` to obtain a Candidate 
+structure from the store, and `saveCandidate(store, candidate)` to save it. Similarly, we use 
+`loadDelegatorBond(store, sender, PubKey)` to load delegator bond with the key (sender and PubKey) from the store, and 
+ `saveDelegatorBond(store, sender, bond)` to save it. `removeDelegatorBond(store, sender, bond)` is used to remove the 
+ bond from the store. 
+     
  
 ### TxDeclareCandidacy
 
-A validator candidacy can be declared using the `TxDeclareCandidacy` transaction.
-During this transaction a self-delegation transaction is executed to bond
-tokens which are sent in with the transaction (TODO: What does this mean?).
+A validator candidacy is declared using the `TxDeclareCandidacy` transaction.
```

Comment:
you don't want to mention that a candidate effectively also has a self delegation? I think this concept is important - it informs others how a candidate can send more tokens to themselves/withdraw